sessiontxn/isolation/readcommitted.go

```diff
@@ -230,14 +284,52 @@ func (p *PessimisticRCTxnContextProvider) AdviseWarmup() error {
 	return nil
 }
 
+func planSkipGetTsoFromPD(sctx sessionctx.Context, plan plannercore.Plan, inLockOrWriteStmt bool) bool {
+	switch v := plan.(type) {
+	case *plannercore.PointGetPlan:
+		return sctx.GetSessionVars().RcPointLockReadUseLastTso && (v.Lock || inLockOrWriteStmt)
+	case plannercore.PhysicalPlan:
+		if len(v.Children()) == 0 {
+			return false
+		}
+		_, isPhysicalLock := v.(*plannercore.PhysicalLock)
+		for _, p := range v.Children() {
+			if !planSkipGetTsoFromPD(sctx, p, isPhysicalLock || inLockOrWriteStmt) {
+				return false
+			}
+		}
+		return true
+	case *plannercore.Update:
+		return planSkipGetTsoFromPD(sctx, v.SelectPlan, true)
+	case *plannercore.Delete:
+		return planSkipGetTsoFromPD(sctx, v.SelectPlan, true)
+	case *plannercore.Insert:
+		// RcInsertUseLastTso controls whether to make a tso request, but not to whether to wait tso requst.
```

Comment:
Is something like `insert ... on duplicate key update` affected?

Reply:
I'll add a test case.